kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/date/instant.kt

```diff
@@ -37,6 +39,17 @@ fun between(fromInstant: Instant, toInstant: Instant) = object : Matcher<Instant
    }
 }
 
+fun closeTo(anotherInstant: Instant, point: Long, unit: ChronoUnit) = object : Matcher<Instant> {
```

Comment:
What do you think about using a duration here?
so,

closeTo(instant, 2.seconds)

Reply:
I understand the intention of your review.
If you modify the function based on your example, it will be as follows.

**closeTo(instant, 2L.toDuration(DurationUnit.SECONDS))** 
_cause **"2L.seconds"** style has been Deprecated_

```kotlin
fun closeTo(anotherInstant: Instant, duration: Duration) = object : Matcher<Instant> {
   override fun test(value: Instant): MatcherResult {
      val range = duration.inWholeNanoseconds
      return MatcherResult(
         abs(value.toEpochMilli() - anotherInstant.toEpochMilli()) <= range,
         { "Expected $value to be close to $anotherInstant in range by $range, but it's not." },
         { "$value is not close to $anotherInstant in range by $range " }
      )
   }
}
```

However, when the prediction fails, I hope that the feedback message reflects my intention to feedback a clear unit and value, not a long-type number. Is there a way to provide that value for the failure feedback message above?

I think it was like this

```kotlin
fun closeTo(anotherInstant: Instant, duration: Duration, unit: DurationUnit) = object : Matcher<Instant> {
   override fun test(value: Instant): MatcherResult {
      val range = duration.inWholeNanoseconds
      return MatcherResult(
         abs(value.toEpochMilli() - anotherInstant.toEpochMilli()) <= range,
         { "Expected $value to be close to $anotherInstant in range by $range$unit, but it's not." },
         { "$value is not close to $anotherInstant in range by $range$unit " }
      )
   }
}
```

But, this does not seem like a good code, especially in terms of redundancy in the Duration Unit.

**closeTo(instant, 2L.toDuration(DurationUnit.SECONDS), DurationUnit.SECONDS)**

Therefore, when using this feature externally, it seems good to hide Duration and focus only on comparison thresholds and unit types. 

like this

**closeTo(instant, 2L, DurationUnit.SECONDS)**

Is there a better solution that includes all my intentions?

Reply:
The 5.seconds style was un-deprecated in Kotlin 1.6 or 1.7, so that's the best way to go I think.
I'd be ok having both styles.

Reply:
I was mistaken., it's un-deprecated 
good, I modified the code [commit](https://github.com/kotest/kotest/commit/6368a19c9a940df8f8943620ed57b50f9fc4cb4c)